### PR TITLE
Improved checking status of control_file_gen.pl, fort.15 and fort.26

### LINKED ISSUE #265: asgs soldiers on inappropriately when it can't find the fort.26 template
case in point

```
[2020-May-29-T20:35:55] INFO: nowcast: control_file_gen.pl: Filling in ADCIRC control template (fort.15).
[2020-May-29-T20:35:55] ERROR: nowcast: control_file_gen.pl: Failed to open the swan template file /worka/work/jgflemin/asgs/input/meshes/LA_v20a/LA_20a-WithUpperAtch.26.template for reading: No such file or directory.
Died at /worka/work/jgflemin/asgs/control_file_gen.pl line 509.
[2020-May-29-T20:36:56-0500] INFO: nowcast:
[2020-May-29-T20:36:56-0500] INFO: asgs_main.sh>checkArchiveFreshness(): Checking to see if the archive of preprocessed subdomain files is up to date.
```
...

etc

btw this manifests in a very unhelpful "Integer divide by zero" error message from padcswan when it can't find the fort.26 file

### PR DESCRIPTION
**Pull request by jasonfleming**
Resolves #265 ; Captures exit code of the `control_file_gen.pl` script as well as missing or zero length fort.15 (and fort.26 if `WAVES=on`). Improves logging of these conditions.

These changes have not been tested yet, @wwlwpd please have a look to see if my approach is sound.

### CODE REVIEW

File: asgs_main.sh [hunk 1]
```diff
@@ -2118,11 +2118,27 @@ if [[ $START = coldstart ]]; then
...
    perl $SCRIPTDIR/control_file_gen.pl $CONTROLOPTIONS >> ${SYSLOG} 2>&1
-   if [ ! -s "$ADVISDIR/$ENSTORM/fort.15" ] ; then
-      echo "hindcast: $ADVISORDIR/$ENSTORM/fort.15 file is 0-length.  This is terminal."  >> ${SYSLOG} 2>&1
-      RMQMessage "EXIT" "$CURRENT_EVENT" "$THIS" "FAIL" "hindcast: fort.15 file is 0-length.  This is terminal."
+   controlExitStatus=$?
+   if [[ $controlExitStatus != 0 ]]; then
+      controlMsg="The control_file_gen.pl script failed with the following error code: '$controlExitStatus'."
+   fi
+   if [[ ! -e $controlFile || ! -s $controlFile ]]; then
+      controlExitStatus=1
```
Comment (wwlwpd) [inline asgs_main.sh:2130]: What is the purpose of changing this `controlExitStatus`? Is this just an error code?

Comment (jasonfleming) [reply asgs_main.sh:2130]: `controlExitStatus` is kind of doing double duty. The actual exit status is used to construct the initial error message, if the exit status is other than zero. After that, `controlExitStatus` is just a flag to indicate whether the perl script exited with an error. The purpose is to maintain troubleshooting granularity between the perl script exiting with an error, and any issues with the resulting `fort.15` and/or `fort.26`. Open to suggestions on improving this.

**Comment by jasonfleming** [comment]
Testing this now. If it runs through a full initialize/nowcast/forecast cycle, I will consider it ready to merge.

File: asgs_main.sh (hunk 1 above)
Comment (jasonfleming) [reply asgs_main.sh:2130]: I think it is working ok, although it could be improved.

**Comment by jasonfleming** [comment]
This ran through a complete cycle, and is working as expected, at least in cases where `control_file_gen.pl` succeeded. Please have a look and merge if it seems ok.

**Review by wwlwpd — APPROVED** [review]
Looks good to me.

**Pull request merged**